src/coreclr/jit/gentree.h

```diff
@@ -2232,6 +2242,22 @@ struct GenTree
         gtFlags &= ~GTF_ICON_HDL_MASK;
     }
 
+#if defined(TARGET_XARCH) && defined(FEATURE_HW_INTRINSICS)
+    bool IsEmbMaskOp()
+    {
+        bool result = (gtFlags & GTF_HW_EM_OP) != 0;
+        assert(!result || (gtOper == GT_HWINTRINSIC));
```

Comment:
This assert is invalid: the same bit used by `GTF_HW_EM_OP` is used by other flags for other GenTree types (e.g., GTF_OVERFLOW).

You could have:
```
assert(gtOper == GT_HWINTRINSIC);
```
if you _only_ allow HWINTRINSIC nodes to call this. Or, you need it to be dynamic:
```
if (gtOper == GT_HWINTRINSIC)
{
    result = (gtFlags & GTF_HW_EM_OP) != 0;
}
else
{
    result = false;
}
```

Reply:
Fixed with https://github.com/dotnet/runtime/pull/98306